service-model-generator/modules/service-model-generator-ls-extension/src/main/java/io/ballerina/servicemodelgenerator/extension/ServiceModelGeneratorService.java

```diff
@@ -0,0 +1,792 @@
+                Optional<Listener> listener = getListenerByName(listenerName.get());
+                if (listener.isEmpty()) {
+                    return new ListenerFromSourceResponse();
+                }
+                Listener listenerModel = listener.get();
+                updateListenerModel(listenerModel, listenerNode);
+                return new ListenerFromSourceResponse(listenerModel);
+            } catch (Exception e) {
+                return new ListenerFromSourceResponse(e);
+            }
+        });
+    }
+
+    @JsonRequest
+    public CompletableFuture<TriggerResponse> getTriggerModel(TriggerRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            if (expectsTriggerByName(request)) {
+                return new TriggerResponse(getTriggerBasicInfoByName(request.packageName()).orElse(null));
+            }
+
+            TriggerProperty triggerProperty = triggerProperties.get(request.id());
+            if (triggerProperty == null) {
+                return new TriggerResponse();
+            }
+            return new TriggerResponse(getTriggerBasicInfoByName(triggerProperty.name()).orElse(null));
+        });
+    }
+
+    @JsonRequest
+    public CompletableFuture<CommonSourceResponse> addFunction(FunctionRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                List<TextEdit> edits = new ArrayList<>();
+                Path filePath = Path.of(request.filePath());
+                Project project = this.workspaceManager.loadProject(filePath);
+                Optional<Document> document = this.workspaceManager.document(filePath);
+                if (document.isEmpty()) {
+                    return new CommonSourceResponse();
+                }
+                SyntaxTree syntaxTree = document.get().syntaxTree();
+                ModulePartNode modulePartNode = syntaxTree.rootNode();
+                TextDocument textDocument = syntaxTree.textDocument();
+                LineRange lineRange = request.codedata().getLineRange();
+                int start = textDocument.textPositionFrom(lineRange.startLine());
+                int end = textDocument.textPositionFrom(lineRange.endLine());
+                NonTerminalNode node = modulePartNode.findNode(TextRange.from(start, end - start), true);
+                if (node.kind() != SyntaxKind.SERVICE_DECLARATION) {
+                    return new CommonSourceResponse();
+                }
+                ServiceDeclarationNode serviceNode = (ServiceDeclarationNode) node;
+                LineRange functionLineRange = serviceNode.openBraceToken().lineRange();
+                NodeList<Node> members = serviceNode.members();
+                if (!members.isEmpty()) {
+                    functionLineRange = members.get(members.size() - 1).lineRange();
+                }
+                String functionNode = "\n\t" + getFunction(request.function()).replace(System.lineSeparator(),
```

Comment:
Shall we replace `\n` with System new line character?

Reply:
Fixed with [1f42fea](https://github.com/ballerina-platform/ballerina-dev-tools/pull/528/commits/1f42fead3729eb90cd2a1d32cb49d6ee61c40453)